runtime/jvmti/jvmtiHelpers.c

```diff
@@ -171,15 +171,19 @@ disposeEnvironment(J9JVMTIEnv * j9env, UDATA freeData)
 
 		j9env->flags |= J9JVMTIENV_FLAG_DISPOSED;
 
+#if JAVA_SPEC_VERSION >= 11
+		J9JVMTI_DATA_FROM_VM(vm)->flags &= ~J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED;
```

Comment:
This clears the flag unconditionally.  What if this environment wasn't the one that had the capability?

Reply:
Good catch, only the environment with that capability can clear it, will fix it.

Reply:
Added a check for current `j9env->capabilities.can_generate_sampled_object_alloc_events`.

In addition, also fixed a check within `jvmtiAddCapabilities()` which should examine incoming `capabilities_ptr` instead of `newCapabilities` which includes existing ones as well.

Please review again.